sim/hunter/wyvern_strike.go

```diff
@@ -0,0 +1,96 @@
+package hunter
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+	"github.com/wowsims/sod/sim/core/stats"
+)
+
+func (hunter *Hunter) getWyvernStrikeConfig(rank int) core.SpellConfig {
+	spellId := [4]int32{0, 458436, 458481, 458482}[rank]
+	dotDamageCoeff := [4]float64{0, 3, 4, 6}[rank]
+	manaCost := [4]float64{0, 55, 75, 100}[rank]
+	level := [4]int{0, 1, 50, 60}[rank]
+
+	hasCobraStrikes := hunter.pet != nil && hunter.HasRune(proto.HunterRune_RuneChestCobraStrikes)
+
+	spellConfig := core.SpellConfig{
+		ActionID:      core.ActionID{SpellID: spellId},
+		SpellSchool:   core.SpellSchoolPhysical,
+		DefenseType:   core.DefenseTypeMelee,
+		ProcMask:      core.ProcMaskMeleeMHSpecial,
+		Flags:         core.SpellFlagMeleeMetrics | core.SpellFlagAPL,
+		Rank:          rank,
+		RequiredLevel: level,
+
+		ManaCost: core.ManaCostOptions{
+			FlatCost: manaCost * (1 - 0.02*float64(hunter.Talents.Efficiency)),
+		},
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+			CD: core.Cooldown{
+				Timer:    hunter.NewTimer(),
+				Duration: time.Second * 8,
+			},
+		},
+		ExtraCastCondition: func(sim *core.Simulation, target *core.Unit) bool {
+			return hunter.DistanceFromTarget <= 5
+		},
+
+		DamageMultiplier: 1,
+		BonusCoefficient: 1,
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "WyvernStrike" + hunter.Label + strconv.Itoa(rank),
+				Tag:   "WyvernStrike",
+			},
+			NumberOfTicks: 8,
+			TickLength:    time.Second * 1,
+
+			OnSnapshot: func(sim *core.Simulation, target *core.Unit, dot *core.Dot, isRollover bool) {
+				dotDamage := dotDamageCoeff / 100 * 8 * hunter.GetStat(stats.AttackPower)
```

Comment:
For the `8 * hunter.GetStat(stats.AttackPower)` bit you can use the `BonusCoefficient` field inside of the `DotConfig` 

```
NumberOfTicks: 8,
TickLength: time.Second * 1,
BonusCoefficient: 8
```